### DISCUSSION TITLE
how do you pass a dict as a parameter to <%namespace:def

### DISCUSSION THREAD

**Discussion by jjgalvez** [sqlalchemy/mako#396] [Usage Questions]
How do I pass a dict to a def when it's called from <%namespace:def>.
I would expect that calling 
```<%namespace:def param="${ {'id':123} }"></namespace:def>``` to work, but when I do I get an error saying I am missing a closing ```}```
for reference ```<%call expr="namespace.def(param={'id':123})"></%call>``` works as expected.

**Comment by zzzeek** [comment]
the brackets probably can't be parsed in that context.  either assign the dict ahead of time in a `<% %>` block or perhaps use `dict(id=123)` form instead.

**Reply by jjgalvez** [reply]
Assigning the dict ahead of time in <% %> works, I was just hoping for a more elegant solution. Also I really can't use dict(id=123) which doesn't work either, because the key is x-on:click which needs to be passed as a string.  Either way the work around works well enough, so thanks for confirming.
